### ISSUE TITLE
linuxdcpp-1.1.0-8: scons fails with: ModuleNotFoundError: No module named 'commands'

### ISSUE THREAD

**Issue by cooljeanius** [fink/fink-distributions#1096]
I'm not too familiar with `scons`, so I dunno how much of this output is relevant:
```
/usr/bin/patch -p1 < /opt/sw/fink/dists/stable/main/finkinfo/net/linuxdcpp.patch
patching file dcpp/CID.h
patching file dcpp/HashValue.h
patching file dcpp/stdinc.h
patching file dcpp/CryptoManager.cpp
/usr/bin/patch -p1 < /opt/sw/fink/dists/stable/main/finkinfo/net/linuxdcpp-clang11-version.patch
patching file SConstruct
Hunk #5 succeeded at 269 (offset 7 lines).
/usr/bin/2to3-2.7 --write --fix=print SConstruct
RefactoringTool: Refactored SConstruct
--- SConstruct	(original)
+++ SConstruct	(refactored)
@@ -8,7 +8,7 @@
 try:
 	from bzrlib import branch
 except ImportError:
-	print "bzrlib not installed"
+	print("bzrlib not installed")
 
 EnsureSConsVersion(0, 98, 1)
 
@@ -49,7 +49,7 @@
 		or (string.atoi(ret[0]) > major)):
 			retval = 1
 	except ValueError:
-		print "No C++ compiler found!"
+		print("No C++ compiler found!")
 
 	context.Result(retval)
 	return retval
@@ -62,7 +62,7 @@
 		b = branch.Branch.open('.')
 		revision = str(b.revno())
 	except:
-		print "failed"
+		print("failed")
 
 	context.env['BZR_REVISION'] = revision
 	context.Result(revision)
@@ -125,7 +125,7 @@
 if os.environ.has_key('CXX'):
 	env['CXX'] = os.environ['CXX']
 else:
-	print 'CXX env variable is not set, attempting to use g++'
+	print('CXX env variable is not set, attempting to use g++')
 	env['CXX'] = 'g++'
 
 if os.environ.has_key('CC'):
@@ -196,28 +196,28 @@
 if not 'install' in COMMAND_LINE_TARGETS:
 
 	if not conf.CheckPKGConfig():
-		print '\tpkg-config not found.'
+		print('\tpkg-config not found.')
 		Exit(1)
 
 	if not conf.CheckPKG('gtk+-2.0 >= 2.12'):
-		print '\tgtk+ >= 2.12 not found.'
-		print '\tNote: You might have the lib but not the headers'
+		print('\tgtk+ >= 2.12 not found.')
+		print('\tNote: You might have the lib but not the headers')
 		Exit(1)
 
 	if not conf.CheckPKG('gthread-2.0 >= 2.4'):
-		print '\tgthread >= 2.4 not found.'
-		print '\tNote: You might have the lib but not the headers'
+		print('\tgthread >= 2.4 not found.')
+		print('\tNote: You might have the lib but not the headers')
 		Exit(1)
 
 	if not conf.CheckPKG('libglade-2.0 >= 2.4'):
-		print '\tlibglade-2.0 >= 2.4 not found.'
-		print '\tNote: You might have the lib but not the headers'
+		print('\tlibglade-2.0 >= 2.4 not found.')
+		print('\tNote: You might have the lib but not the headers')
 		Exit(1)
 
 	
 	if not conf.CheckCXXHeader('boost/version.hpp', '<>'):
-		print '\tboost not found.'
-		print '\tNote: You might have the lib but not the headers'
+		print('\tboost not found.')
+		print('\tNote: You might have the lib but not the headers')
 		Exit(1)
 
 	if not conf.CheckHeader('time.h'):
@@ -230,34 +230,34 @@
 		Exit(1)
 
 	if not conf.CheckLibWithHeader('pthread', 'pthread.h', 'c'):
-		print '\tpthread library not found'
-		print '\tNote: You might have the lib but not the headers'
+		print('\tpthread library not found')
+		print('\tNote: You might have the lib but not the headers')
 		Exit(1)
 
 	if not conf.CheckLibWithHeader('z', 'zlib.h', 'c'):
-		print '\tz library (gzip/z compression) not found'
-		print '\tNote: You might have the lib but not the headers'
+		print('\tz library (gzip/z compression) not found')
+		print('\tNote: You might have the lib but not the headers')
 		Exit(1)
 
 	if not conf.CheckLibWithHeader('bz2', 'bzlib.h', 'c'):
-		print '\tbz2 library (bz2 compression) not found'
-		print '\tNote: You might have the lib but not the headers'
+		print('\tbz2 library (bz2 compression) not found')
+		print('\tNote: You might have the lib but not the headers')
 		Exit(1)
 
 	# This needs to be before ssl check on *BSD systems
 	if not conf.CheckLib('crypto'):
-		print '\tcrypto library not found'
-		print '\tNote: This library may be a part of libssl on your system'
+		print('\tcrypto library not found')
+		print('\tNote: This library may be a part of libssl on your system')
 		Exit(1)
 
 	if not conf.CheckLibWithHeader('ssl', 'openssl/ssl.h', 'c'):
-		print '\tOpenSSL library (libssl) not found'
-		print '\tNote: You might have the lib but not the headers'
+		print('\tOpenSSL library (libssl) not found')
+		print('\tNote: You might have the lib but not the headers')
 		Exit(1)
 
 	# Needed for XFlush(). Headers shouldn't be needed since we include gdk/gdkx.h
 	if not conf.CheckLib('X11'):
-		print '\tX11 library not found'
+		print('\tX11 library not found')
 		Exit(1)
 
 	if not conf.CheckHeader('iconv.h'):
@@ -271,8 +271,8 @@
 	# TODO: Implement a plugin system so libnotify doesn't have compile-time dependencies
 	if conf.env.get('libnotify'):
 		if not conf.CheckPKG('libnotify >= 0.4.1'):
-			print '\tlibnotify >= 0.4.1 not found, disabling notifications.'
-			print '\tNote: You might have the lib but not the headers'
+			print('\tlibnotify >= 0.4.1 not found, disabling notifications.')
+			print('\tNote: You might have the lib but not the headers')
 		else:
 			conf.env.Append(CPPDEFINES = 'HAVE_LIBNOTIFY')
 			conf.env.ParseConfig('pkg-config --libs libnotify')
RefactoringTool: Files that were modified:
RefactoringTool: SConstruct
sudo -u fink-bld [ENV] sh -c /tmp/fink.TTEuu
/tmp/fink.hpG1l
#!/bin/sh -ev
export CPATH=/opt/sw/include
/opt/sw/bin/scons PREFIX=/opt/sw $MAKEFLAGS
scons: Reading SConscript files ...
ModuleNotFoundError: No module named 'commands':
  File "/opt/sw/src/fink.build/linuxdcpp-1.1.0-8/linuxdcpp-1.1.0/SConstruct", line 5:
    import commands
### execution of /tmp/fink.hpG1l failed, exit code 2
### execution of /tmp/fink.TTEuu failed, exit code 2
Removing runtime build-lock...
Removing build-lock package...
/opt/sw/bin/dpkg-lockwait -r fink-buildlock-linuxdcpp-1.1.0-8
(Reading database ... 
dpkg: serious warning: files list file for package `fink-buildlock-ccp4-6.5-1' missing, assuming package has no files currently installed.
1030811 files and directories currently installed.)
Removing fink-buildlock-linuxdcpp-1.1.0-8 ...
```
Possibly a python version incompatibility? cc @nieder 
(I'm on x86_64 Big Sur with Xcode 13, as usual)

**Comment by nieder** [comment]
Can you retry? I don't have a system to test it on because one machine is way behind big sur and another is way ahead.

**Comment by cooljeanius** [comment]
OK so after [6d53cfd](https://github.com/fink/fink-distributions/commit/6d53cfd584320a99f2f5ccc6753665c650f5f7f2), `fink` now no longer thinks that linuxdcpp needs to be upgraded? Is that the desired outcome?

**Comment by nieder** [comment]
You're on BigSur, which is macOS 11, and I just restricted it to 10.9-11.0, so revision -8 should be visible to you. What's the output of `fink -V` ?

**Comment by cooljeanius** [comment]
> You're on BigSur, which is macOS 11, and I just restricted it to 10.9-11.0, so revision -8 should be visible to you. What's the output of `fink -V` ?

```
$ fink -V
Package manager version: 0.45.99.git
Distribution version: selfupdate-git Wed Oct 25 08:05:08 2023, 11.3, x86_64
Trees: local/main stable/main local/injected

Copyright (c) 2001 Christoph Pfisterer
Copyright (c) 2001-2019 The Fink Package Manager Team
This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
```

**Comment by nieder** [comment]
This should be fixed by 7c00bbcbfb14a6f73b8e6004a53b708c61d03d88 I'd forgotten that we had to special case macOS 11.3-11.7

**Comment by cooljeanius** [comment]
> This should be fixed by [7c00bbc](https://github.com/fink/fink-distributions/commit/7c00bbcbfb14a6f73b8e6004a53b708c61d03d88) I'd forgotten that we had to special case macOS 11.3-11.7

OK I tried updating, but `fink` didn't seem to be able to find a decent mirror to download `scons3` from...

**Comment by cooljeanius** [comment]
> > This should be fixed by [7c00bbc](https://github.com/fink/fink-distributions/commit/7c00bbcbfb14a6f73b8e6004a53b708c61d03d88) I'd forgotten that we had to special case macOS 11.3-11.7
> 
> OK I tried updating, but `fink` didn't seem to be able to find a decent mirror to download `scons3` from...

...ok, I can confirm that ed750ad seems to have fixed this; thanks!
